kuma/search/utils.py

```diff
@@ -72,13 +75,17 @@ def search_exception_handler(exc):
     # to get the standard error response.
     response = exception_handler(exc)
 
-    if (response is None and
-            isinstance(exc, elasticsearch.ElasticsearchException)):
-        # FIXME: This really should return a 503 error instead but Zeus
-        # doesn't let that through and displays a generic error page in that
-        # case which we don't want here
-        log.error('Elasticsearch exception: %s' % exc)
-        return Response({'error': SEARCH_DOWN_DETAIL},
-                        status=status.HTTP_200_OK)
+    if response is None:
+        if isinstance(exc, elasticsearch.ElasticsearchException):
+            # FIXME: This really should return a 503 error instead but Zeus
+            # doesn't let that through and displays a generic error page in that
+            # case which we don't want here
+            log.error('Elasticsearch exception: %s' % exc)
+            return Response({'detail': SEARCH_DOWN_DETAIL},
+                            status=status.HTTP_200_OK)
+        elif isinstance(exc, UnicodeDecodeError):
+            log.error('UnicodeDecodeError exception: %s' % exc)
+            return Response({'detail': SEARCH_ERROR_DETAIL},
+                            status=status.HTTP_404_NOT_FOUND)
```

Comment:
I've done some research and 404 are more appropriate for this type of issue, wrong parameters.

The code in question *is* handled by this DRF exception handler, there were just two spots where this would show tracebacks.